spec/controllers/books_controller_spec.rb

```diff
@@ -4,52 +4,45 @@
   let!(:books) { create_list(:book, 3) }
 
   describe 'GET #index' do
-    context 'With an unauthenticated user' do
+    context 'with an unauthenticated user' do
       it 'returns an unauthorized status' do
         get :index, params: {}
         expect(response).to have_http_status(:unauthorized)
       end
     end
 
-    context 'With an authenticated user' do
+    context 'with an authenticated user' do
       include_context 'Authenticated User'
 
-      it 'without params returns all books' do
-        get :index
-        expect(JSON.parse(response.body)['total_count']).to eq(books.size)
+      context 'with a correct param' do
+        it 'returns a 200 ok status' do
+          get :index, params: { book: { genre: books.first.genre } }
+          expect(response).to have_http_status(:ok)
+        end
       end
 
-      it 'returns a 200 ok status' do
-        get :index, params: { book: { genre: books.first.genre } }
-        expect(response).to have_http_status(:ok)
+      context 'without params' do
+        it 'returns all books' do
+          get :index
+          expect(JSON.parse(response.body)['total_count']).to eq(books.size)
+        end
       end
 
-      def unique_title
-        rnd_title = Faker::Book.title
-        rnd_title = Faker::Book.title while books.map(&:title).include?(rnd_title)
-        rnd_title
+      context 'with inexistent filters' do
+        it 'returns empty response' do
+          get :index, params: { book: { title: 'HY43Ewcqw!$ds1' } }
+          expect(JSON.parse(response.body)['total_count']).to eq(0)
+        end
       end
 
-      it 'params filters not exists' do
-        get :index, params: { book: { title: unique_title } }
-        expect(JSON.parse(response.body)['total_count']).to eq(0)
-      end
-
-      def full_index_response
-        b = books.first
-        get :index, params: { book: {
-          title: b.title,
-          # Below are additional params
-          image: b.image,
-          publisher: b.publisher,
-          year: b.year
-        } }
-      end
-
-      it 'setting additionals params' do
-        full_index_response
-        expect(response).to have_http_status(:ok)
-        # it response status ok becouse additionals params are not used
+      context 'with additionals params' do
+        it 'additionals params are not used' do
+          b = books.first
+          get :index, params: { book: { title: b.title,
+                                        image: b.image, publisher: b.publisher, year: b.year } }
+          book_response = Book.new(JSON.parse(response.body)['page'][0])
+          expect(book_response.title).to be_eql(books.first.title)
```

Comment:
Maybe we can clean up this a little. Some changes I would do here
```ruby
it 'returns the correct filtered collection' do
  b = books.first
  get :index, params: { book: { title: b.title, image: b.image, publisher: b.publisher, year: b.year } }
  expect(JSON.parse(response.body)['page'].first['tittle']).to eq(b.title)
end
``` 

I would move this example to `with a correct param` context. What do you think?